storage/src/block_meta_storage.rs

```diff
@@ -525,17 +504,20 @@ fn merge_meta_value(
     existing_val: Option<&[u8]>,
     operands: &mut MergeOperands,
 ) -> Option<Vec<u8>> {
+    if let Some(val) = existing_val {
+        if val.len() < LEN_FIXED_META {
+            return None;
+        }
+    }
+
     let mut result = existing_val.map(|v| v.to_vec());
 
     for op in operands {
         match result {
             Some(ref mut val) => {
-                assert!(
```

Comment:
To me it looks like the length constraint should be checked for both existing value and merge operands. Also we cannot do anything apart returning `None` in case of invalid value.